src/ring_theory/ideal/basic.lean

```diff
@@ -623,11 +653,6 @@ theorem coe_subset_nonunits [semiring α] {I : ideal α} (h : I ≠ ⊤) :
   (I : set α) ⊆ nonunits α :=
 λ x hx hu, h $ I.eq_top_of_is_unit_mem hx hu
 
-lemma exists_max_ideal_of_mem_nonunits [comm_semiring α] (h : a ∈ nonunits α) :
-  ∃ I : ideal α, I.is_maximal ∧ a ∈ I :=
-begin
-  have : ideal.span ({a} : set α) ≠ ⊤,
-  { intro H, rw ideal.span_singleton_eq_top at H, contradiction },
-  rcases ideal.exists_le_maximal _ this with ⟨I, Imax, H⟩,
-  use [I, Imax], apply H, apply ideal.subset_span, exact set.mem_singleton a
-end
+lemma mem_nonunits_iff_exists_max_ideal [comm_semiring α] :
+a ∈ nonunits α ↔ ∃ I : ideal α, I.is_maximal ∧ a ∈ I :=
+iff.trans mem_nonunits_iff' ideal.not_has_left_inv_iff_mem_maximal
```

Comment:
The statement after the `:` should be indented
```suggestion
lemma mem_nonunits_iff_exists_max_ideal [comm_semiring α] :
  a ∈ nonunits α ↔ ∃ I : ideal α, I.is_maximal ∧ a ∈ I :=
mem_nonunits_iff'.trans ideal.not_has_left_inv_iff_mem_maximal
```
also, dot notation makes this slightly shorter